src/utils/messageGeneration/messages.js

```diff
@@ -603,11 +603,7 @@ function deserializeInnerMessage(spec, message, buffer, bufferOffset) {
         bufferOffset[0] += 4; // only for variable length arrays
       }
 
-      let ArrayMsgClass;
       let isPrimitive = field.isBuiltin;
-      if (!isPrimitive) {
-        ArrayMsgClass = getMessageFromRegistry(arrayType, 'msg');
```

Comment:
I'm not sure about getting rid of these classes during deserialization, especially since they seem to be class instances otherwise. I'm assuming the main issue here is that we're doing this lookup on every deserialization, so I'd prefer to find a way to only do this once.

Not sure what @chfritz thinks as he has more experience using the on-the-fly messages.

Reply:
yes, I got the same way at first time, but then I found this lookup means nothing, the spec will be got in line 631 spec.getMsgSpecForType(field.baseType), and arrayMessage that passed in , works as empty container.

Reply:
Any clients that are relying on the deserialized format being `instanceof Message` won't be able to... I'd guess this isn't too widespread, but it's odd that we do it on [initialization](https://github.com/RethinkRobotics-opensource/rosnodejs/blob/kinetic-devel/src/utils/messageGeneration/messages.js#L445) of messages.

Reply:
if we new the msg, we do know the msg type already.  I think all we do is to set/get data, like  `msg.data` .    and we still able to `instanceof Message` for whole msg, except child msg

Reply:
yes it's a pretty minor case...

I think I'm ok to leave this as is for now. We can move back to having `Message` instances if needed, but I think we can do it in a more optimal way.